cmd/proxy/wsl_integration_linux.go

```diff
@@ -42,71 +42,42 @@ func main() {
 	flag.BoolVar(&debug, "debug", false, "enable additional debugging.")
 	flag.StringVar(&upstreamAddr, "upstream-addr", k8sAPI, "The upstream server's address (k3s API sever).")
 	flag.StringVar(&listenAddr, "listen-addr", defaultListenAddr, "The server's address in an IP:PORT format.")
+	flag.StringVar(&logFile, "logfile", "/var/log/wsl-proxy.log", "path to the logfile for wsl-proxy process")
 	flag.Parse()
 
-	if debug {
-		logrus.SetLevel(logrus.DebugLevel)
-	}
+	setupLogging(logFile)
 
-	listener, err := net.Listen("tcp", listenAddr)
+	proxy, err := portproxy.NewPortProxy("/run/wsl-proxy.sock")
 	if err != nil {
-		logrus.Fatalf("Failed to listen on %s: %s", listenAddr, err)
+		logrus.Errorf("failed to create listener for published ports: %s", err)
+		return
 	}
 
 	// Handle graceful shutdown
 	sigCh := make(chan os.Signal, 1)
 	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
 
-	// WaitGroup to wait for all connections to finish before shutting down
-	var wg sync.WaitGroup
-
 	go func() {
 		<-sigCh
 		logrus.Println("Shutting down...")
-		listener.Close()
-		wg.Wait()
+		proxy.Close()
```

Comment:
This ignores the error from `Close`; is that okay?

Since we always call `Wait` after `Close`, why not make `Close` include the wait?